### ISSUE TITLE
Extend `gravmag.transform.reduction_to_pole()` to take arrays for inc and dec

### ISSUE THREAD

**Issue by mtb-za** [fatiando/fatiando#422]
I have been doing some additional work on things, and have, in combination with https://github.com/scivision/pyigrf12 got a grid matching my survey block with how the inclination and declination varies across the survey area.

I would be nice to have an additional RTP method that can take such grids and use them when doing the RTP steps.

Is this something other people would see value in me spending some time in doing?

(side note: it would be nice to get a version of pyigrf that does not need a Fortran compiler to be installed, especially for Windows people....)

**Comment by leonfoks** [comment]
I'm surprised that Python is not using its canned Fortran compiler.  If the setup.py file is told about the Fortran code, it should just compile and install for you. 

As for Fortran on windows, I have instructions on how to install the free GNU compilers [here](https://github.com/leonfoks/coretran)  (Sorry the shameless plug)

**Comment by mtb-za** [comment]
Yeah, it complains that I do not have a fortran compiler installed. Not sure if it is on my end or if there is something odd on their end. But that is probably off-topic for here anyway. Just want to gauge interest in extending `inc` and `dec` to optionally also being arrays.

**Comment by leouieda** [comment]
@mtb-za I don't think anyone has actually solved this problem yet. At least I've never seen an RTP with variable inc, dec. I don't know how simple this would be in the Fourier domain. The equations assume constant inc, dec so you might have to derive new ones.

Not sure how useful this would be because RTP of large areas is generally not a good idea because of remanence effects. But it's more "correct" than what we currently do, so I say go for it! If you want to tackle this,  I'd be willing to help. Let me know.

@leonfoks please, more shameless plugs :)

> it would be nice to get a version of pyigrf that does not need a Fortran compiler

We can build conda packages for it so that the user wouldn't need to compile. But it's not exactly painless. 

Another option is to do the synthesis in [SHTOOLS](https://github.com/SHTOOLS/SHTOOLS) using the published coefficients. Though it does need a compiler still.
